spec-tests/src/test/java/org/candlepin/spec/content/ContentResourceSpecTest.java

```diff
@@ -0,0 +1,286 @@
+
+        Object jsonPayload = consumerApi.exportCertificates(consumer.getUuid(), null);
+        List<JsonNode> jsonNodes = CertificateUtil.extractEntitlementCertificatesFromPayload(jsonPayload);
+
+        assertEquals(1, jsonNodes.size());
+        JsonNode jsonBody = jsonNodes.get(0);
+        assertEquals(1, jsonBody.get("products").size());
+        assertEquals(2, jsonBody.get("products").get(0).get("content").size());
+
+        JsonNode contentOutput1;
+        JsonNode contentOutput3;
+        // figure out the order of products in the cert, so we can assert properly
+        String firstContentName = jsonBody.get("products").get(0).get("content")
+            .get(0).get("name").asText();
+        if (firstContentName.equals(content1.getName())) {
+            contentOutput1 = jsonBody.get("products").get(0).get("content").get(0);
+            contentOutput3 = jsonBody.get("products").get(0).get("content").get(1);
+        }
+        else {
+            contentOutput1 = jsonBody.get("products").get(0).get("content").get(1);
+            contentOutput3 = jsonBody.get("products").get(0).get("content").get(0);
+        }
+
+        assertEquals(content1.getType(), contentOutput1.get("type").asText());
+        assertEquals(content1.getName(), contentOutput1.get("name").asText());
+        assertEquals(content1.getLabel(), contentOutput1.get("label").asText());
+        assertEquals(content1.getVendor(), contentOutput1.get("vendor").asText());
+        assertEquals(content1.getContentUrl(), contentOutput1.get("path").asText());
+        assertEquals(content1.getArches(), contentOutput1.get("arches").get(0).asText());
+
+        assertEquals(content3.getType(), contentOutput3.get("type").asText());
+        assertEquals(content3.getName(), contentOutput3.get("name").asText());
+        assertEquals(content3.getLabel(), contentOutput3.get("label").asText());
+        assertEquals(content3.getVendor(), contentOutput3.get("vendor").asText());
+        assertEquals(content3.getContentUrl(), contentOutput3.get("path").asText());
+        assertEquals(0, contentOutput3.get("arches").size());
+    }
+
+    @Test
+    @DisplayName("should filter content with mismatched architecture from the product")
+    public void filterContentWithMismatchedArchitectureFromProduct() throws Exception {
+        OwnerDTO owner = ownerApi.createOwner(Owners.random());
+        // Even though this product has no arches specified, and should normally not be filtered out,
+        // the product it belongs to has an architecture that mismatches with the system's,
+        // so we do expect it to get filtered out.
+        ContentDTO content1 = createContent(owner.getKey(), StringUtil.random("cname1"),
+            StringUtil.random("test-content1"), StringUtil.random("clabel1"), StringUtil.random("ctype1"),
+            StringUtil.random("cvendor1"), "/this/is/the/path", null);
+        AttributeDTO attribute1 = new AttributeDTO();
```

Comment:
Check ProductAttributes/Products. I added enum for easier handling of products and attributes.

Reply:
Awesome! Updated to use Products and ProductAttributes